crates/datasources/src/sqlserver/mod.rs

```diff
@@ -341,6 +350,76 @@ impl TableProvider for SqlServerTableProvider {
     }
 }
 
+/// Convert filtering expressions to a predicate string usable with the
+/// generated SQL Server query.
+fn exprs_to_predicate_string(exprs: &[Expr]) -> Result<String> {
+    let mut ss = Vec::new();
+    let mut buf = String::new();
+    for expr in exprs {
+        if write_expr(expr, &mut buf)? {
+            ss.push(buf);
+            buf = String::new();
+        }
+    }
+
+    Ok(ss.join(" AND "))
+}
+
+/// Try to write the expression to the string, returning true if it was written.
+fn write_expr(expr: &Expr, buf: &mut String) -> Result<bool> {
+    match expr {
+        Expr::Column(col) => {
+            write!(buf, "{}", col)?;
+        }
+        Expr::Literal(val) => {
+            util::encode_literal_to_text(util::Datasource::SqlServer, buf, val)?;
+        }
+        Expr::IsNull(expr) => {
+            if write_expr(expr, buf)? {
+                write!(buf, " IS NULL")?;
+            } else {
+                return Ok(false);
+            }
+        }
+        Expr::IsNotNull(expr) => {
+            if write_expr(expr, buf)? {
+                write!(buf, " IS NOT NULL")?;
+            } else {
+                return Ok(false);
+            }
+        }
+        Expr::IsTrue(expr) => {
+            if write_expr(expr, buf)? {
+                write!(buf, " IS TRUE")?;
+            } else {
+                return Ok(false);
+            }
+        }
+        Expr::IsFalse(expr) => {
+            if write_expr(expr, buf)? {
+                write!(buf, " IS FALSE")?;
+            } else {
+                return Ok(false);
+            }
+        }
```

Comment:
Could we add an SLT with a query that would trigger this? SQL Server doesn't have a bool data type, so I'm not positive this would work.

Reply:
Yeah, I just raised a PR to run tests in CI (some issue with local). I will keep resolving stuff accordingly. There are tests in `basic.slti` that will validate this.